garden_ai/containers.py

```diff
@@ -30,14 +33,24 @@ def start_container_with_notebook(
       and is exposed to the host on the same port.
     - The token for accessing the notebook is still the JUPYTER_TOKEN variable.
     """
-    with console.status(f"[bold green] Pulling image: {base_image}"):
-        client.images.pull(base_image, platform="linux/x86_64")
+    # first check if the image is present, otherwise attempt to pull
+    try:
+        client.images.get(base_image)
+    except docker.errors.ImageNotFound:
+        with console.status(f"[bold green] Pulling image: {base_image}"):
+            client.images.pull(base_image, platform="linux/x86_64")
```

Comment:
This seems like a sensible change to me. @OwenPriceSkelly is more in the thick of it so I'll get his take for a second opinion

Reply:
I thought this was already the default behavior of a naive docker pull but I could be mistaken! what kind of case would this behave differently? if it's equivalent I'd just as soon leave it

Reply:
The impetus was that before I implemented the change, I got an error that I was not permitted to pull from `gardenai`, since the default image name is `gardenai/base` and of course I don't have such a permission.

Reply:
I think that's because `gardenai/base` doesn't exist yet, not a permission thing -- should have defaulted to `gardenai/test` which is hopefully public, if you still can't pull that one I need to fix that

Reply:
I guess it then becomes this: we need our default image names to be things that theoretically could be pulled (which feels counterintuitive to me, since we are doing so knowing that the image has no meaning and will never actually be pulled), or we only attempt to pull when the image is not already present.
Reading what I wrote, my bias is very clear, but I hope it doesn't take away from the binary.